lib/matplotlib/axis.py

```diff
@@ -1635,6 +1635,11 @@ def set_ticklabels(self, ticklabels, *, minor=False, **kwargs):
         locator = (self.get_minor_locator() if minor
                    else self.get_major_locator())
         if isinstance(locator, mticker.FixedLocator):
+            if len(locator.locs) != len(ticklabels):
```

Comment:
L 1620 probably needs something like "Number of labels must be the same as number of ticks set by `set_ticks` or the `FixedLocator`"?

Reply:
I added a reference to set_ticks.

Reply:
That helps, but I mean above in the Parameter description.  Its a bit unclear as to what is meant by "tick labels".   maybe:

```
ticklabels : sequence of str or of `.Text`\s
            List of texts for tick labels, one for each tick defined in `.axis.set_ticks`.
```